File: fabric-audit-agent/core/ticket.js
import { domainOf } from './key-utils.js';

/**
 * Build a tracker-agnostic work item from a finding. Pure.
 * @param {object} finding
 * @returns {{ title:string, body:string, severity:string, labels:string[], externalKey:string|undefined }}
 */
export function buildTicket(finding = {}) {
  const level = finding.score?.level ?? 'Info';
  const fixes = (finding.fix ?? []).map(x => `- ${x}`).join('\n');
  return {
    title: `[${level}] ${finding.what ?? 'Fabric audit finding'}`,
    body: [
      `Where: ${finding.where ?? ''}`,
      `Why: ${finding.why ?? ''}`,
      `Impact: ${finding.impact ?? ''}`,
      `Fix:\n${fixes}`,
    ].join('\n\n'),
    severity: level,
    labels: ['fabric-audit', domainOf(finding.key)],
    externalKey: finding.key,
  };
}

File: fabric-audit-agent-py/fabric_audit_agent/ticket.py
"""Build a tracker-agnostic work item from a finding. Port of ``core/ticket.js``. Pure."""
from .key_utils import domain_of


def build_ticket(finding=None):
    finding = finding or {}
    level = (finding.get("score") or {}).get("level") or "Info"
    fixes = "\n".join(f"- {x}" for x in (finding.get("fix") or []))
    return {
        "title": f"[{level}] {finding.get('what') or 'Fabric audit finding'}",
        "body": "\n\n".join([
            f"Where: {finding.get('where') or ''}",
            f"Why: {finding.get('why') or ''}",
            f"Impact: {finding.get('impact') or ''}",
            f"Fix:\n{fixes}",
        ]),
        "severity": level,
        "labels": ["fabric-audit", domain_of(finding.get("key"))],
        "externalKey": finding.get("key"),
    }
